src/Framework/FactoryResolverAwareTrait.php

```diff
@@ -5,11 +5,24 @@
 namespace Gacela\Framework;
 
 use Gacela\Framework\ClassResolver\Factory\FactoryResolver;
+use RuntimeException;
 
+/**
+ * @method static AbstractFactory factory()
+ */
 trait FactoryResolverAwareTrait
 {
     private ?AbstractFactory $factory = null;
 
+    public static function __callStatic(string $name, array $arguments): AbstractFactory
```

Comment:
Does it make sense to call `parent::__callStatic($name, $arguments)` before doing our own logic?

Reply:
wow! I didn't think about this 🤔
I don't know, actually, do you see any potential problem if we don't have this parent call?

Reply:
It pops to my mind because we had already some problems with Laravel in the past, the magic methods like `__call`, `__callStatic` and so on are special methods, and it's easier to get conflicts with other frameworks/user-defined classes if we don't take that into account... 👀 
But cannot sentence anything at the moment.

We can try as it is for now, and if someone complains in the future or we spot some issue, we can change it 👍🏼